### PR TITLE
MonitoredExecutionContext

### PR DESCRIPTION
**Pull request by stkem**
Random idea to gain more visibility into waiting futures/blocked threads as well as limit concurrency at the point of use (so that a specific task type can't overwhelm the thread pool alone). Not sure if it's a good idea/worth the overhead.
RFC @eishay @andrewconner

### CODE REVIEW

**Comment by stkem** [comment]
- @ymatsuda @raymondng

**Comment by stkem** [comment]
We could have a plugin that periodically (not _too_ frequently) checks the execution context and sends stats to statsd.

File: kifi-backend/modules/common/app/com/keepit/common/akka/MonitoredExecutionContext.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  def waitingCount : Int = waiting.get()
+  def runningCount : Int = running.get()
+
+  def execute(runnable: Runnable): Unit = {
+    waiting.incrementAndGet()
+    val wrapped = new Runnable {
+      def run() = {
+        waiting.decrementAndGet()
+        running.incrementAndGet()
+        runnable.run()
```
Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/common/akka/MonitoredExecutionContext.scala]: do we need something like

``` scala
try 
  runnable.run() 
finally 
  running.decrementAndGet()
```

?

Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/akka/MonitoredExecutionContext.scala]: Ah, yes. I believe we do. Good catch.

**Pull request closed without merging**